File: Find The Peak/UnitTests.java
import org.junit.Test;
import static org.junit.Assert.*;

public class UnitTests{
    @Test
    public void test1(){
        FindThePeak finder = new FindThePeak();
        int[] nums = {1,2,3,1};
        int index = finder.findPeakElement(nums);
        assertTrue(index == 2);
    }

    @Test
    public void test2(){
        FindThePeak finder = new FindThePeak();
        int[] nums = {1,2,1,3,5,6,4};
        int index = finder.findPeakElement(nums);
        assertTrue(nums[index] > (index > 0 ? nums[index - 1] : Integer.MIN_VALUE));
        assertTrue(nums[index] > (index < nums.length - 1 ? nums[index + 1] : Integer.MIN_VALUE));
    }

    @Test
    public void test3(){
        FindThePeak finder = new FindThePeak();
        int[] nums = {42};
        int index = finder.findPeakElement(nums);
        assertEquals(0, index);
    }

    @Test
    public void test4(){
        FindThePeak finder = new FindThePeak();
        int[] nums = {1, 2};
        int index = finder.findPeakElement(nums);
        assertTrue(index == 1);
    }

    @Test
    public void test5(){
        FindThePeak finder = new FindThePeak();
        int[] nums = {5,4,3,2,1};
        int index = finder.findPeakElement(nums);
        assertEquals(0, index);
    }
}

File: Find The Peak/unit_tests.py
import unittest
from find_peak import Solution

class UnitTests(unittest.TestCase):
    def stUp(self):
        self.sol = Solution()

    def test1(self):
        nums = [1, 2, 3, 1]
        result = self.sol.findPeakElement(nums)
        self.assertIn(result, [2])

    def test2(self):
        nums = [1,2,1,3,5,6,4]
        result = self.sol.findPeakElement(nums)
        self.assertIn(result, [1,5])  # 2 and 6 are both peaks

    def test3(self):
        nums = [10]
        result = self.sol.findPeakElement(nums)
        self.assertEqual(result, 0)

    def test4(self):
        nums = [1,2]
        result = self.sol.findPeakElement(nums)
        self.assertEqual(result, 1)

    def test5(self):
        nums = [5,4,3,2,1]
        result = self.sol.findPeakElement(nums)
        self.assertEqual(result, 0)

    def test6(self):
        nums = [1,2,3,4,5]
        result = self.sol.findPeakElement(nums)
        self.assertEqual(result, 4)

if __name__ == "__main__":
    unittest.main()